### ISSUE TITLE
Hi, I am going to share some data with you guys.

### ISSUE THREAD

**Issue by ValorLin** [ShishirPatil/gorilla#99]
Hi, I am interested in this project and I have done some reasearch. I tried the evaluation on the TorchHub API senerio with some of the latest models and other fine-tuned models.
* The gorilla-ada/babbage/curie/davinci are fine-tuned models base on OpenAI's ada/babbage/curie/davinci models. Trained with the data in this project (torchhub_train.json) and got suprising result. They are much better, the gorilla-babbage model gets a 90.86% accuracy.
* The latest gpt-3.5-turbo get 29.57% improvement compared to gpt-3.5-turbo-0301 and gpt-4 get 22.05% improvment compared to gpt-4-0315.
Here is the result:
<img width="945" alt="image" src="https://github.com/ShishirPatil/gorilla/assets/3257035/5ed1d0f8-e46a-43a1-9246-dce349479e63">
<img width="958" alt="image" src="https://github.com/ShishirPatil/gorilla/assets/3257035/fba40ada-55b6-4bdb-8388-2212cd16cbb2">

Below is a example of the training data that I used for fine-tuning on OpenAI models.
```
{"prompt":"I need an API that can estimate the relative depth of objects from a single image. The output should be an inverse depth map.\n\n###\n\n","completion":" {'domain': 'Classification', 'api_call': \"model = torch.hub.load('intel-isl\/MiDaS', 'DPT_Hybrid', pretrained=True)\", 'api_provider': 'PyTorch', 'explanation': \"Load the pretrained MiDaS model for estimating relative depth from a single image, which outputs an inverse depth map.\", 'code': 'import torch\nmodel = torch.hub.load(\"intel-isl\/MiDaS\", \"DPT_Hybrid\", pretrained=True)'}###"}
```
The full converted training data can be found here: https://gist.github.com/ValorLin/807ac5f75361775dfd2cc31f21a03e77

I hope that data is helpful. Its very interesting result, you should try it.

**Comment by ShishirPatil** [comment]
Hey @ValorLin thank's for sharing this! Super insightful. Yeah, one thing we might want to keep in mind is that OpenAI's models are much larger ~175B-ish, vs the 7B Gorilla models - so the performance is quite understandable. BTW do you mind writing a blog or a write-up, with your pipeline including the data-generation, results, insights on where it does well, not so well, etc ! Would be a great read for the community - including and especially me 😄  

Nice job!
